src/bin/miri.rs

```diff
@@ -233,6 +233,9 @@ fn main() {
                 "-Zmiri-ignore-leaks" => {
                     miri_config.ignore_leaks = true;
                 }
+                "-Zmiri-panic-on-unsupported-syscalls" => {
```

Comment:
"syscall" isn't really the right name here; what Miri is intercepting are C-level foreign function calls.

Maybe `-Zmiri-panic-on-unsupported`; that would also leave the door open for using this for more things in the future?